### ISSUE TITLE
[URL Matching by priority] inconsistent result in firefox

### ISSUE THREAD

**Issue by aitboudad** [ui-router/core#59]
Given the following routes:
```ts
{ url: '/', name: 'home', component: ... }
{ url: '/{page}', name: 'cms', component: ... }
```
when visiting the home page in firefox the url match `cms` rule first but it should be 'home'

Workaround:

```patch
- { url: '/{page}', name: 'cms', component: ... }
+ { url: '/{page}', name: 'cms', params: { page: { value: '' } }, }
```

**Comment by christopherthielen** [comment]
Can you reliably reproduce it? Is this ONLY an issue in firefox?

This plunker loaded in firefox loads the home page:
http://plnkr.co/edit/n2Yqav7g9QgDc7CJG66k?p=preview

**Comment by aitboudad** [comment]
Sorry it's not an issue on firefox but the order of registration if I set `cms` state before` home`, I expect the `home` to be resolved first when there is no params http://plnkr.co/edit/66UJM9m0kgospI8TUvqK?p=preview ?

**Comment by christopherthielen** [comment]
This was fixed in 5.0.6 by https://github.com/ui-router/core/commit/73a1fe0
